src/env.h

```diff
@@ -625,7 +628,22 @@ class Environment {
   // List of id's that have been destroyed and need the destroy() cb called.
   inline std::vector<double>* destroy_async_id_list();
 
-  std::unordered_multimap<int, loader::ModuleWrap*> module_map;
+  class ModuleGlobalHash {
+   public:
+    explicit ModuleGlobalHash(v8::Isolate* isolate) : isolate_(isolate) {}
+    size_t operator()(const v8::Global<v8::Module>& module) const {
+      return module.Get(isolate_)->GetIdentityHash();
+    }
+
+   private:
+    v8::Isolate* isolate_;
+  };
+
+  std::unordered_map<int, loader::ModuleWrap*> id_to_module_wrap_map;
+  std::unordered_map<
```

Comment:
can't this have collisions? We should probably ensure that even if there is a collision we can have the duplicates properly stored. Perhaps use a `multimap` here?